### DISCUSSION TITLE
map tile area flashing partly or wholly black fairly often when app is idle (no touch events) on iOS

### DISCUSSION THREAD

**Discussion by diffent** [react-native-maps/react-native-maps#4461] [Q&A]
### Summary

we are using react-native-maps 0.24.2 and are unable to update to the latest version quickly due to version compat issues with other node modules and cocoapod modules and react-native version etc.  we noticed the above issue on newer iOS versions (15+).  if any one is seeing this issue, we found a work around to take a view snapshot in the tile rendering complete callback, place the CALayer of the snapshot in the proper place in the layer hierarchy, then remove the snapshot when we get the tile rendering start callback.  it is a hack but is the best we could do for now since we dont have access to the lower level google maps rendering code and cannot update to newer versions to see if the problem occurs.  all changes are in file react-native-maps/lib/ios/AirGoogleMaps/AIRGoogleMapManager.m ... we use a CALayer insertion rather than a UIView insertion to avoid blocking touch events.  we have attached snaps of changes in case someone needs this fix, but we have not set up a pull request since this is an old version of maps and are unsure if this may have other implications.  the map maintainers may have a better idea.   
 
<img width="1468" alt="mappatch1" src="https://user-images.githubusercontent.com/470335/192013327-c5506ac2-b433-4955-b484-bf57adaf8fdc.png">
<img width="1490" alt="mappatch2" src="https://user-images.githubusercontent.com/470335/192013337-8337eb26-2427-4e19-8af3-eb767fc4f1b8.png">

### Reproducible sample code

```TSX
see additional information for the patches
```

### Steps to reproduce

map is idle on screen.  wait for a minute or less.  you will start to see random flashes of black as the map tiles are background rendered.  sometimes some or all of the view is stuck on black.  it sometimes fixes itself.  when you touch the map, it redraws and fixes itself.  

### Expected result

no black flashing in map area.

### Actual result

black flashing, stuck black in map area.

### React Native Maps Version

0.24.2

### What platforms are you seeing the problem on?

iOS (Google Maps)

### React Native Version

0.60.5

### What version of Expo are you using?

Not using Expo

### Device(s)

iphone X (ios 15.latest), iphone 12 mini (ios 16), iphone se (old) ios 15.latest

### Additional information

all changes from react-native-maps/lib/ios/AirGoogleMaps/AIRGoogleMapManager.m

this is not the minimal set of changes, some prior fix trials are left in 

new lines are marked with // patch

@interface AIRGoogleMapManager() <GMSMapViewDelegate, GMSIndoorDisplayDelegate>
{
  BOOL didCallOnMapReady;
  UIView * debugMapView; // patch
  CALayer * snaplayer;   // patch
}
@end

// this is a new method entirely

-(void)mapViewSnapshotReady:(GMSMapView *)mapView {
   //[mapView stopRendering];
   static UIView * snapview = nil;
   
   [snapview removeFromSuperview];
   
   snapview = [mapView snapshotViewAfterScreenUpdates:(BOOL)YES];
   snapview.tag = 'snap';
   
   CGRect f = snapview.frame;
   
   f.origin.x = 0;
   f.origin.y = 0;
   
   snapview.frame = f;
   
   //[mapView addSubview:snapview];
   
   [snaplayer removeFromSuperlayer];
   
   snaplayer = snapview.layer;
   
   [mapView.layer addSublayer:snaplayer];
   
}

// this is an existing method, add 1 line patch

- (void)mapViewDidStartTileRendering:(GMSMapView *)mapView {
  AIRGoogleMap *googleMapView = (AIRGoogleMap *)mapView;
  [googleMapView didPrepareMap];
  [snaplayer removeFromSuperlayer]; // patch
}

**Comment by monholm** [comment]
Hi @diffent 👋

Thank you for providing this information for anyone interested. I've moved this to the discussion section, as it's not a current bug :)

**Reply by erquhart** [reply]
@monholm would you mind expanding on your comment a bit? Maps black out after a few seconds of idle time on iOS, and this happens pretty reliably. Why would it not be considered a current bug?
